main.js

```diff
@@ -0,0 +1,226 @@
+  .then(json => {
+    console.log({lat: json[0].lat, lon: json[0].lon});
+    // assing the map variable, set it to id of html div
+    // and set the view to lat and lon of restaurant
+    var map = L.map('mapid').setView([json[0].lat, json[0].lon], 16);
+  
+    // add tile layer to map, from Interactive Maps slides, change later?
+    L.tileLayer('https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
+      attribution: 'Map data &copy; <a href="https://www.openstreetmap.org/">OpenStreetMap</a> contributors'
+    }).addTo(map);
+
+    // if there is an address passed into the function
+    if(address !== 'Burlington, VT 05401') {
+      // set marker on the coordinates of the restaurant
+      L.marker([json[0].lat, json[0].lon]).addTo(map)
+      .on('click', () => document.location.href = website);
+        
+    // otherwise, address is Burlington so display all the markers
+    } else {
+      // pass map so function knows where to put markers
+      displayAllMarkers(map);
+    }
+  });
+}
+
+// takes the name of a restaurant and fetches the json
+// file for that restaurant, 
+async function getAddress(name) {
+  //if no name is passed as an argument
+  if (!name) {
+    console.log("no place name specfied");
+    // return B. VT 05401 as the name of the place
+    return 'Burlington, VT 05401'
+  } else {
+    console.log("fetching place named '" + name + "'");
+  
+    //fetch restaurant json as a Promise, wait for it to finish
+    const response = await fetch(name + '.json');
+    // convert response to json, wait for it to finsih
+    const object = await response.json();
+    //get address and return
+    const address = object.address;
+    console.log(address);
+    return address;
+    }
+}
+
+// takes the name of a restaurant, fetches the json
+// file for that restaurant, and returns the website 
+async function getWebsite(name) {
+  //if no name is passed as an argument
+  if (!name) {
+    console.log("no place name specfied");
+    // return index.html because there is no website
+    return 'index.html';
+  } else {
+    console.log("fetching website for '" + name + "'");
+  
+    //fetch restaurant json as a Promise, wait for it to finish
+    const response = await fetch(name + '.json');
+    // convert response to json, wait for it to finsih
+    const object = await response.json();
+    //get website and return
+    const website = object.website;
+    console.log(website);
+    return website;
+    }
+}
+
+// function handles showing all of the markers
+// on the map when an address is not specified
+// uses async/await method to fetch json
+async function displayAllMarkers(map) {
+  console.log('were gonna display all the markers');
+  // create an array to hold to lats and longs of all restaurants
+  let latsLongs = [];
+
+  // fetch all restaurant names & get json, wait to finish
+  let response = await fetch('all.json');
+  let json = await response.json();
+  let address;
```

Comment:
You should be able to just define these three variables inside your `for ... of` loop since you don't access them anywhere but inside that loop. It would clean your code up a bit, and make sure you're not accidentally redefining those variables elsewhere.